pandas/io/formats/excel.py

```diff
@@ -299,6 +300,24 @@ def _border_style(self, style: str | None, width: str | None, color: str | None)
             if width_name in ("hair", "thin"):
                 return "dashed"
             return "mediumDashed"
+        elif style in (
+            "hair",
+            "mediumDashDot",
+            "dashDotDot",
+            "mediumDashDotDot",
+            "dashDot",
+            "slandDashDot",
+            "mediumDashed",
+        ):
+            # Excel-specific styles
+            return style
+        else:
+            warnings.warn(
+                f"Unhandled border style format: {repr(style)}",
```

Comment:
Wouldn't it be better to raise so the user can specify a supported style?

Reply:
I believe the current code tends to prefer issuing `CSSWarning` instead of raising, such as in `CSSToExcelConverter.color_to_excel`.

From personal experience, it can be frustrating if the export code fails after doing expensive loading/manipulations to get to my final DataFrame, and then having to fix the error and repeat all over again. My thought is to export the file anyways and let the user decide if they want to rerun/reexport, or just clean up the formatting manually in Excel.